### PR TITLE
Crash fix

### LINKED ISSUE #532: Crash on tryToInit
**Describe the bug**
App crashes on Android 6 devices. It misses deleteSharedPreferences method.
Do instead 
```
getSharedPreferences(name, mode).edit().clear().commit()
```

**SDK Version**
com.walletconnect:sign:2.1.0
com.walletconnect:android-core:1.3.0

**To Reproduce**
Run on Android 6

**Expected behavior**
No Crash

image

```
Exception java.lang.NoSuchMethodError: No virtual method deleteSharedPreferences(Ljava/lang/String;)Z in class Landroid/content/Context; or its super classes (declaration of 'android.content.Context' appears in /system/framework/framework.jar)
  at com.walletconnect.android.impl.di.StorageModuleKt$$InternalSyntheticApiModelOutline$1$6ab672f0e040553a6a4fab4b5b949f164b5c97312491f9122c092f47f2b28a40$0.m (StorageModuleKt.java)
  at com.walletconnect.android.internal.common.di.CryptoModuleKt$androidApiCryptoModule$1.invoke$deleteSharedPreferences (CryptoModule.kt:37)
  at com.walletconnect.android.internal.common.di.CryptoModuleKt$androidApiCryptoModule$1.access$invoke$deleteSharedPreferences (CryptoModule.kt:17)
  at com.walletconnect.android.internal.common.di.CryptoModuleKt$androidApiCryptoModule$1$1.invoke (CryptoModule.kt:46)
  at com.walletconnect.android.internal.common.di.CryptoModuleKt$androidApiCryptoModule$1$1.invoke (CryptoModule.kt:41)
  at org.koin.core.instance.InstanceFactory.create (InstanceFactory.kt:54)
  at org.koin.core.instance.SingleInstanceFactory.create (SingleInstanceFactory.kt:46)
  at org.koin.core.instance.SingleInstanceFactory$get$1.invoke (SingleInstanceFactory.kt:53)
  at org.koin.core.instance.SingleInstanceFactory$get$1.invoke (SingleInstanceFactory.kt:51)
  at org.koin.mp.KoinPlatformTools.synchronized (KoinPlatformTools.kt:20)
  at org.koin.core.instance.SingleInstanceFactory.get (SingleInstanceFactory.kt:51)
  at org.koin.core.registry.InstanceRegistry.resolveInstance$koin_core (InstanceRegistry.kt:111)
  at org.koin.core.scope.Scope.resolveValue (Scope.kt:255)
  at org.koin.core.scope.Scope.resolveInstance (Scope.kt:242)
  at org.koin.core.scope.Scope.get (Scope.kt:205)
  at org.koin.core.scope.Scope.get$default (Scope.java:131)
  at com.walletconnect.android.internal.common.di.CryptoModuleKt$androidApiCryptoModule$1$2.invoke (CryptoModule.kt:51)
  at com.walletconnect.android.internal.common.di.CryptoModuleKt$androidApiCryptoModule$1$2.invoke (CryptoModule.kt:51)
  at org.koin.core.instance.InstanceFactory.create (InstanceFactory.kt:54)
  at org.koin.core.instance.SingleInstanceFactory.create (SingleInstanceFactory.kt:46)
  at org.koin.core.instance.SingleInstanceFactory$get$1.invoke (SingleInstanceFactory.kt:53)
  at org.koin.core.instance.SingleInstanceFactory$get$1.invoke (SingleInstanceFactory.kt:51)
  at org.koin.mp.KoinPlatformTools.synchronized (KoinPlatformTools.kt:20)
  at org.koin.core.instance.SingleInstanceFactory.get (SingleInstanceFactory.kt:51)
  at org.koin.core.registry.InstanceRegistry.resolveInstance$koin_core (InstanceRegistry.kt:111)
  at org.koin.core.scope.Scope.resolveValue (Scope.kt:255)
  at org.koin.core.scope.Scope.resolveInstance (Scope.kt:242)
  at org.koin.core.scope.Scope.get (Scope.kt:205)
  at org.koin.core.scope.Scope.get$default (Scope.java:131)
  at com.walletconnect.android.internal.common.di.CryptoModuleKt$androidApiCryptoModule$1$3.invoke (CryptoModule.kt:53)
  at com.walletconnect.android.internal.common.di.CryptoModuleKt$androidApiCryptoModule$1$3.invoke (CryptoModule.kt:53)
  at org.koin.core.instance.InstanceFactory.create (InstanceFactory.kt:54)
  at org.koin.core.instance.SingleInstanceFactory.create (SingleInstanceFactory.kt:46)
  at org.koin.core.instance.SingleInstanceFactory$get$1.invoke (SingleInstanceFactory.kt:53)
  at org.koin.core.instance.SingleInstanceFactory$get$1.invoke (SingleInstanceFactory.kt:51)
  at org.koin.mp.KoinPlatformTools.synchronized (KoinPlatformTools.kt:20)
  at org.koin.core.instance.SingleInstanceFactory.get (SingleInstanceFactory.kt:51)
  at org.koin.core.registry.InstanceRegistry.resolveInstance$koin_core (InstanceRegistry.kt:111)
  at org.koin.core.scope.Scope.resolveValue (Scope.kt:255)
  at org.koin.core.scope.Scope.resolveInstance (Scope.kt:242)
  at org.koin.core.scope.Scope.get (Scope.kt:205)
  at org.koin.core.scope.Scope.get$default (Scope.java:131)
  at com.walletconnect.android.relay.RelayClient.initialize (RelayClient.kt:42)
  at com.trustwallet.walletconnect.client.v2.WCClientLazyInitializer.tryToInit (WCClientLazyInitializer.kt:34)
  at com.wallet.crypto.trustapp.App.onCreate (App.kt:100)
  at android.app.Instrumentation.callApplicationOnCreate (Instrumentation.java:1037)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:6428)
  at android.app.ActivityThread.access$1800 (ActivityThread.java:229)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1887)
  at android.os.Handler.dispatchMessage (Handler.java:102)
  at android.os.Looper.loop (Looper.java:148)
  at android.app.ActivityThread.main (ActivityThread.java:7331)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1230)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1120)
```

### PR DESCRIPTION
**Pull request by vcoolish**
Fixes #532

### CODE REVIEW

File: androidCore/impl/src/release/kotlin/com.walletconnect.android.impl.di/StorageModule.kt [hunk 1]
```diff
@@ -53,7 +54,10 @@ private fun Scope.createSharedPreferences(): SharedPreferences {
 }
 
 private fun Scope.deleteSharedPreferences() {
-    androidContext().deleteSharedPreferences(SHARED_PREFS_FILENAME)
+    androidContext()
+        .getSharedPreferences(SHARED_PREFS_FILENAME, Context.MODE_PRIVATE)
+        ?.edit()
+        ?.commit()
```
Comment (Elyniss) [inline androidCore/impl/src/release/kotlin/com.walletconnect.android.impl.di/StorageModule.kt:60]: Missed a `clear()` call

`getSharedPreferences(name, mode).edit().clear().commit()`

**Review by Elyniss — CHANGES_REQUESTED** [review]

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/CryptoModule.kt [hunk 2]
```diff
@@ -33,9 +33,10 @@ fun androidApiCryptoModule() = module {
     }
 
     fun Scope.deleteSharedPreferences() {
-        if (androidContext().getSharedPreferences(sharedPrefsFile, Context.MODE_PRIVATE) != null) {
-            androidContext().deleteSharedPreferences(sharedPrefsFile)
-        }
+        androidContext()
+            .getSharedPreferences(sharedPrefsFile, Context.MODE_PRIVATE)
+            ?.edit()
+            ?.commit()
```
[the remark this answers is not in the document]
Comment (vcoolish) [reply androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/CryptoModule.kt:39]: oh thanks!

**Comment by Elyniss** [comment]
Previous functionality removed the sharedPreferences altogether. Since `clear()` does not remove I suggest we also delete file when API version is lower than 7.0

Like this:
```
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            return context.deleteSharedPreferences(name)
        } else {
            context.getSharedPreferences(name, MODE_PRIVATE).edit().clear().apply()
            val dir = File(context.applicationInfo.dataDir, "shared_prefs")
            return File(dir, "$name.xml").delete()
        }
```

**Review by jakobuid — APPROVED** [review]
Thanks @vcoolish for contribution and quick action!

**Review by Elyniss — APPROVED** [review]

**Pull request merged**